### PR TITLE
ENG-3865 Run all microservices

### PR DESCRIPTION
**Pull request by alepintus**

### CODE REVIEW

File: src/commands/run.ts [hunk 1]
```diff
@@ -15,12 +20,31 @@ export default class Run extends BaseBuildCommand {
...
   public async run(): Promise<void> {
     BundleService.isValidBundleProject()
-    const { args } = await this.parse(Run)
+    const { args, flags } = await this.parse(Run)
     const componentService = new ComponentService()
 
-    if (args.name === undefined) {
+    if (Object.keys(flags).length > 0 && args.name !== undefined) {
+      this.error(
+        `Run failed, please use one flag or write the component name as argument`,
```
Comment (cecchisandrone) [inline src/commands/run.ts]: Bad arguments. Please use the component name as argument or one of the available flags, not both

File: src/commands/run.ts [hunk 2]
```diff
@@ -15,12 +20,31 @@ export default class Run extends BaseBuildCommand {
...
+        { exit: 1 }
+      )
+    }
+
+    if (flags['all-ms']) {
+      CliUx.ux.action.start(
+        `Running all microservices. Press ctrl + c to exit.`
+      )
+      await this.runAllComponents(Phase.Run, ComponentType.MICROSERVICE)
+    }else {
```
Comment (cecchisandrone) [inline src/commands/run.ts]: bad formatting?

File: src/commands/run.ts [hunk 3]
```diff
@@ -15,12 +20,31 @@ export default class Run extends BaseBuildCommand {
...
+     if (args.name === undefined) {
       this.error(`Run failed, missing required arg name`, { exit: 1 })
```
Comment (cecchisandrone) [inline src/commands/run.ts]: Bad arguments. Please use the component name as argument or one of the available flags

File: src/commands/run.ts [hunk 4]
```diff
@@ -45,6 +69,32 @@ export default class Run extends BaseBuildCommand {
...
 
-    CliUx.ux.action.stop()
+    }
   }
+
+  public async runAllComponents(
+    commandPhase: Phase,
+    componentType?: ComponentType
+  ): Promise<void> {
+    this.log(color.bold.blue('Run components...'))
```
Comment (cecchisandrone) [inline src/commands/run.ts]: Running components `component1`, `component2`

File: src/utils.ts [hunk 5]
```diff
@@ -11,3 +13,35 @@ export class InMemoryWritable extends Writable {
     next()
   }
 }
+
+export class OutputColorWritable extends Writable {
```
Comment (cecchisandrone) [inline src/utils.ts]: maybe better `ColorizedWritable`?

Comment (alepintus) [reply src/utils.ts]: done

File: src/commands/run.ts (hunk 4 above)
Comment (alepintus) [reply src/commands/run.ts]: I added the component type , I suggest to address this change (add the list of components) in another task

File: src/commands/run.ts (hunk 1 above)
Comment (alepintus) [reply src/commands/run.ts]: The user can pass 3 flags as well, so I generalized the message to "Bad arguments. Use the component name as argument or one of the available flags" I also refactored a bit the code to have the same error messages for the build command

File: src/commands/run.ts (hunk 2 above)
Comment (alepintus) [reply src/commands/run.ts]: Reformatted with prettier

File: src/commands/run.ts (hunk 3 above)
Comment (alepintus) [reply src/commands/run.ts]: done

**Review by cecchisandrone — APPROVED** [review]

**Pull request merged**